db/migration/1716401298509-AddTagGraph.ts

```diff
@@ -0,0 +1,55 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class AddTagGraph1716401298509 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`-- sql
+            CREATE TABLE tag_graph (
+                parentId INT NOT NULL,
+                childId INT NOT NULL,
+                weight INT NOT NULL DEFAULT 1,
+                PRIMARY KEY (parentId, childId),
+                FOREIGN KEY (parentId) REFERENCES tags (id),
+                FOREIGN KEY (childId) REFERENCES tags (id),
+                INDEX (childId),
+                CONSTRAINT chk_no_self_link CHECK (parentId != childId)
+            )`)
+
+        // create root tag
+        await queryRunner.query(`-- sql
+            INSERT INTO tags (name) VALUES ("tag-graph-root")`)
```

Comment:
This leads to tag-graph-root being a valid tag in various selection dropdowns. Probably not a big deal since our users are not out to sabotage our system but maybe nice to eventually filter it out